packages/interaction-kit/src/application.ts

```diff
@@ -150,80 +147,44 @@ export default class Application {
 		return undefined;
 	}
 
-	loadApplicationCommandDirectory(directory: string) {
-		console.log(`Loading Application Commands from ${directory}`);
-
-		fs.readdir(directory, async (err, files) => {
-			if (err) {
-				throw err;
-			}
-
-			console.log(`\tLoading ${files.length} files`);
-			for (const file of files) {
-				if (file.endsWith(".js")) {
-					const command = (await import(path.join(directory, file))) as Module<
-						InteractionKitCommand<ApplicationCommandInteraction>
-					>;
-					this.addCommand(command.default);
-				}
-			}
-		});
-
-		return this;
-	}
-
-	loadMessageComponentDirectory(directory: string) {
-		console.log(`Loading Message Components from ${directory}`);
-
-		fs.readdir(directory, async (err, files) => {
-			if (err) {
-				throw err;
-			}
-
-			console.log(`\tLoading ${files.length} files`);
-			for (const file of files) {
-				if (file.endsWith(".js")) {
-					const component = (await import(
-						path.join(directory, file)
-					)) as Module<SerializableComponent>;
-					this.addComponent(component.default);
-				}
-			}
-		});
-
-		return this;
-	}
-
-	async handler(event: FetchEvent) {
+	async handler(request: Request) {
 		console.log("REQUEST");
 
-		if (event.request.method !== "POST") {
-			await event.respondWith(
-				response(ResponseStatus.MethodNotAllowed, { error: "Invalid Method" })
-			);
-			return;
+		if (request.method !== "POST") {
+			return response(ResponseStatus.MethodNotAllowed, {
+				error: "Invalid Method",
+			});
 		}
 
-		const valid = await isValidRequest(event.request, this.#publicKey);
+		const valid = await isValidRequest(request, this.#publicKey);
 		if (!valid) {
-			await event.respondWith(
-				response(ResponseStatus.Unauthorized, { error: "Invalid Request" })
-			);
-			return;
+			return response(ResponseStatus.Unauthorized, {
+				error: "Invalid Request",
+			});
 		}
 
 		try {
-			const json = (await event.request.json()) as APIInteraction;
-			void Interaction.handler(
-				this,
-				json,
-				async (status: ResponseStatus, json: Record<string, any>) => {
-					void event.respondWith(response(status, json));
-				}
-			);
+			const json = await request.json<APIInteraction>();
+			console.log({ json });
+			return await new Promise((resolve) => {
+				void Interaction.handler(
+					this,
+					json,
+					(status: ResponseStatus, json: Record<string, any>) => {
+						console.log("responding");
+						console.log(status);
+						console.log(JSON.stringify(json));
+						resolve(response(status, json));
```

Comment:
Can this be cleaned up?

Reply:
if you weren't so attached to callbacks you could, you're promisifying callbacks instead of writing real async functions to begin with